GCC 14 reports -Wstringop-overflow when serializing a dynamic byte vector

With zpp_bits v4.7.2 and GCC 14.2, serializing a value followed by a dynamic byte vector produces a warning:

zpp_bits.h:2182:28: warning: ‘void* memcpy(void*, const void*, size_t)’ writing 3 or more bytes into a region of size 2 overflows the destination [-Wstringop-overflow=]
note: at offset 5 into destination object of size 7 allocated by ‘operator new’

Minimal reproduction: https://godbolt.org/z/8o6osce7G

Codex claims, that this is a false positive because enlarge_for() resizes the destination before the copy. For us it's still an issue because it sounds scary and we want to compile with warnings as errors :)

--- a/zpp_bits.h
+++ b/zpp_bits.h
@@ -2178,6 +2178,7 @@ protected:
 #if !defined __clang__ && defined __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overflow"
 #endif
                 std::memcpy(m_data.data() + m_position,
                             item.data(),

